system-addon/lib/TelemetryFeed.jsm

```diff
@@ -274,6 +280,37 @@ this.TelemetryFeed = class TelemetryFeed {
   }
 
   /**
+   * setPageModeForPerf - Set page mode for the session perf object. A page
+   * could be loaded in a combination of following modes: {
+   *   "regular": 0,
+   *   "preloaded": 1 << 0,
+   *   "prerendered": 1 << 1
+   * }
+   *
+   * Note:
+   *   1. Preloading doesn't apply to about:home.
+   *   2. The first newtab page is not preloaded, nor prerendered. This function
+   *      does not take that into account provided that it should not skew the
+   *      global measurement.
+   *
+   * @param {obj} session A session object
+   */
+  setPageModeForPerf(session) {
+    let pageMode = 0;
+
+    if (this._prefs.get("prerender")) {
+      pageMode |= PAGE_MODES_ENCODING.prerendered;
```

Comment:
This gets called at the session end where the user might have changed something to have `prerender` pref change. The pref check should be fine when the page is loading though.

Reply:
Good catch. Since this function relies on the other two fields from `session.perf`, perhaps the best place to call it is inside `saveSessionPerfData`.

Reply:
Yep, I like this idea!